### ISSUE TITLE
Feature request: customize resize grab target size separately from border size

### ISSUE THREAD

**Issue by ethus3h** [compiz-reloaded/emerald#19]
When my theme has very thin borders, I'd like to be able to make the resize target larger than the border size.

**Comment by kn00tcn** [comment]
we need this everywhere (xfcewm, mate decorator, gtk2)

i think i've seen gtk3 themes control it on gnome3 apps (ones that dont have the window border theme of emerald or xfce or mate)

**Comment by moontide** [comment]
There's another method to resize window: in `ccsm`, enable `Resize Window` plugin, then you can use <kbd>ALT</kbd> + <kbd>BUTTON2</kbd> (middle mouse button) to resize window. It's easier than grab the corner of window, I actually like this method.

**Comment by kn00tcn** [comment]
yes, several DEs have a hotkey based resize toggle, i prefer to have the mouse so i can be lazy

technically i dont mind seeing the borders either, tiny ones that some themes have make it hard to see where each window is if all of them are text with a white background for example

**Comment by alekseyt** [comment]
Does this look good or not? Note mouse cursor near bottom-left corner.

![emerald grabs](https://user-images.githubusercontent.com/993454/35483336-5f6a0af4-0449-11e8-99fd-40211106e852.png)

Defaults are all 0 (zeroes).

I tried to make minimal changes and preserve full backward compatibility. But some stuff could not be preserved, for example magic numbers like -8. Nonetheless, in my situation it works seemingly in the same way as before, but borderless windows can now be resized.

Backward compatibility and minimal changes also means that some bugs transitioned too. For example bottom side grab breaks after vertical resize and fixes itself after horizontal resize. I'm sure someone can open another issue on this if needed, right?

I can make it even more minimal, but i formatted some code for readability and left something in comments for history.

If you want to try this, i linked commit to this issue (above this comment). Of course everything can be moved around and linked commit is probably not the last one related to this issue. Please let me know if there are any issues with this.
